contrib/src/main/java/gov/nasa/jpl/aerie/contrib/models/SampledResource.java

```diff
@@ -1,35 +1,68 @@
 package gov.nasa.jpl.aerie.contrib.models;
 
 import gov.nasa.jpl.aerie.merlin.framework.resources.discrete.DiscreteResource;
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
 
 import java.util.Objects;
 import java.util.function.Supplier;
 import java.util.function.UnaryOperator;
 
 import static gov.nasa.jpl.aerie.merlin.framework.ModelActions.*;
-import static gov.nasa.jpl.aerie.merlin.protocol.types.Duration.SECOND;
 
+/**
+ * Simple resource that samples arbitrarily many existing resources/values at a specified rate (default rate is once per
+ * second).
+ */
 public class SampledResource<T> implements DiscreteResource<T> {
   private final Register<T> result;
   private final Supplier<T> sampler;
+  private double rate;
 
+  /**
+   * Constructor that does not require caller to specify a rate and therefore assumes a sample
+   * rate of 1 sample per second.
+   */
   public SampledResource(final Supplier<T> sampler, final UnaryOperator<T> duplicator) {
```

Comment:
I like how you preserved backwards compatibility here :+1: That makes this update straightforward for anyone currently relying on this.

That said - your change makes this class more useful - I have my doubts that it was getting heavy use before.

Reply:
Daemon tasks -- especially threaded ones -- will incur some pretty decent task overhead, especially at high sampling rates. Ideally, this should be a tool of last resort.

Reply:
While I don't disagree with you, I will also say that as a new "modeler" to Aerie, this resource was a tool of first resort because it is easy to understand and easy to use. Moreover, there aren't many other out of the box resources that I can grab and use that are simple to define. Without a nice collection of first-line solution resources at a modelers disposal, they will likely be lazy like me and use what they have available to them.

Reply:
I agree with @ewferg as it comes to our API, we need to make the easy thing, the right thing.

Reply:
No immediate action here, so I will close this out